nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi-env.sh

```diff
@@ -16,16 +16,38 @@
 #    limitations under the License.
 #
 
+# By default this file will unconditionally override whatever environment variables you have set
+# and set them to defaults defined here.
+# If you want to define your own versions outside of this script please set the environment variable
+# NIFI_OVERRIDE_NIFIENV to "true". That will then use whatever variables you used outside of
+# this script.
+
 # The java implementation to use.
 #export JAVA_HOME=/usr/java/jdk1.8.0/
 
-export NIFI_HOME=$(cd "${SCRIPT_DIR}" && cd .. && pwd)
+setOrDefault() {
+  declare envvar="$1" default="$2"
+
+  local res="$envvar"
+  if [ -z "$envvar" ] || [ "$NIFI_OVERRIDE_NIFIENV" != "true" ]
```

Comment:
Also it would cause problems with variables that has a meaning when set to an empty string, because that would be changed to the default value. 

For posix shell we could use `(: "${somevar?}")` to check for the variable beeing set, but I beleive that should be done before calling the `setOrDefault` method.

I'm not sure we have to be that careful, but at least if there is a good option to cover that use case we should do it.

Reply:
Didn't mean to block this. I recommend showing a warning message (probably to stderr) when setting a default value for a variable.

Reply:
Sorry @pepov ! No worries at all. I'm on vacation and won't be back for another week. I'll get to it then. Your comments make sense. I just haven't gotten around to look at them in detail.

Reply:
Well..the vacation turned out to be much longer than expected ;-)
But I'm back on this PR and would like to get it fixed so I'll work on an updated version now.